cmd/admin-handlers-users.go

```diff
@@ -2816,7 +2816,7 @@ func commonAddServiceAccount(r *http.Request, ldap bool) (context.Context, auth.
 	denyOnly := (targetUser == cred.AccessKey || targetUser == cred.ParentUser)
 	if ldap && !denyOnly {
 		res, _ := globalIAMSys.LDAPConfig.GetValidatedDNForUsername(targetUser)
-		if res.NormDN == cred.ParentUser {
+		if res != nil && res.NormDN == cred.ParentUser {
```

Comment:
Open Q: Is this the behaviour we want in case nil is returned @donatello

Reply:
@klauspost we are not checking the returned error here so this is expected that it will crash.

Reply:
yeah that is another implementation issue with GetValidatedDNForUsername() there should be a way to communicate perhaps it is generally a good idea to avoid `return nil, nil` it is complex to handle.